packages/story-utils/src/generateSnapshots.ts

```diff
@@ -27,20 +27,17 @@ function getDefaultSnapshot(wrapper: RenderResult) {
  * @param storyFileExports the exports of a .stories.tsx file including the default export with additional context
  * @param options the options for the test to add necessary context specific to snapshot tests
  */
-export default function generateSnapshots<
-  S extends StoryFileExports<Args>,
-  Args,
->(
-  storyFileExports: S,
-  { getSnapshot = getDefaultSnapshot, argOverrides }: TestOptions<Args> = {},
+export default function generateSnapshots(
+  storiesImport: StoriesImport,
+  { getSnapshot = getDefaultSnapshot }: TestOptions = {},
 ): void {
-  const stories = getStoriesFromStoryFileExports(storyFileExports);
+  const stories = composeStories(storiesImport);
```

Comment:
Using `@storybook/testing-react` here to do story collecting/processing now, which

- Makes our code simpler since we don't have to worry about it.
- Removes the need to enforce specific patterns in stories (such as not defining args at the top level), since `@storybook/testing-react` just handles everything correctly.